server/cpho/views/indicator_data.py

```diff
@@ -133,21 +145,52 @@ def formset(self):
                 dimension_type=self.dimension_type
             )
 
+        literal_possible_values = existing_data.filter(
+            literal_dimension_val__isnull=False
+        )
+        if self.dimension_type is not None:
+            literal_possible_values = literal_possible_values.filter(
+                dimension_type=self.dimension_type
+            )
+
         instances = []
         for pv in possible_values:
             if existing_data_by_dimension_value.get(pv, None):
                 record = existing_data_by_dimension_value[pv]
             else:
                 record = IndicatorDatum(
-                    indicator=self.indicator, dimension_value=pv
+                    indicator=self.indicator,
+                    dimension_value=pv,
+                    dimension_type=self.dimension_type,
+                )
+            instances.append(record)
+
+        for nlpv in literal_possible_values:
+            if existing_data_by_literal_value.get(
+                nlpv.literal_dimension_val, None
+            ):
+                print("found")
+                record = existing_data_by_literal_value[
+                    str(nlpv.literal_dimension_val)
+                ]
+            else:
+                print("not found")
+                record = IndicatorDatum(
+                    indicator=self.indicator,
+                    dimension_value=None,
+                    dimension_type=self.dimension_type,
+                    literal_dimension_val=nlpv.literal_dimension_val,
```

Comment:
I would think this block never gets called, right?

Reply:
i think you're right!
ill comment out the else block and test it
:)